cmd/kubectl_token_oauth.go

```diff
@@ -84,15 +85,24 @@ func rancherLogin(input *LoginInput, provider TypedProvider, oauthToken *oauth2.
 		return nil, err
 	}
 
-	b, err := request(http.MethodPost, url, bytes.NewBuffer(jsonBody))
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(jsonBody))
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+
+	resp, respBody, err := doRequest(client, req)
+	if err == nil && resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected http status code %d", resp.StatusCode)
+	}
```

Comment:
Same as above, can we invert the check?

Reply:
This was a bad copy-paste from one of the my previous iterations. We shouldn't invert anything. All we need is to set the err. Fixed in 649c29df36cec21939cc05ce66114d8e4d11cd97.